homeassistant/components/tolosauna/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for tolosauna."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from tololib import ToloClient
+import voluptuous as vol
+
+from homeassistant.components.dhcp import IP_ADDRESS, MAC_ADDRESS
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DEFAULT_NAME, DEFAULT_RETRY_COUNT, DEFAULT_RETRY_TIMEOUT, DOMAIN
+
+DATA_SCHEMA_USER = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ToloSaunaConfigFlow(ConfigFlow, domain=DOMAIN):
+    """ConfigFlow for TOLO Sauna."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize ToloSaunaConfigFlow."""
+        self._discovered_host: str | None = None
+
+    @staticmethod
+    def _check_device_availability(host: str) -> bool:
+        client = ToloClient(host)
+        result = client.get_status_info(
+            resend_timeout=DEFAULT_RETRY_TIMEOUT, retries=DEFAULT_RETRY_COUNT
+        )
+        return result is not None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+
+            device_available = await self.hass.async_add_executor_job(
+                self._check_device_availability, user_input[CONF_HOST]
```

Comment:
`_check_device_availability` can raise, which isn't catched

Reply:
Wow, I'm impressed. How did you find this?

Check will be added in next commit.